x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcArray.java

```
@@ -0,0 +1,99 @@
+            return getArray();
+        }
+        throw new SQLFeatureNotSupportedException("getArray with non-empty Map not supported");
+    }
+
+    @Override
+    public Object getArray(long index, int count) throws SQLException {
+        if (index < 1 || index > Integer.MAX_VALUE) {
+            throw new SQLException("Index value [" + index + "] out of range [1, " + Integer.MAX_VALUE + "]");
+        }
+        if (count < 0) {
+            throw new SQLException("Illegal negative count [" + count + "]");
+        }
+        int index0 = (int) index - 1; // 0-based index
+        int available = index0 < values.size() ? values.size() - index0 : 0;
+        return available > 0 ?
+            Arrays.copyOfRange(values.toArray(), index0, index0 + Math.min(count, available)):
+            new Object[0];
+    }
+
+    @Override
+    public Object getArray(long index, int count, Map<String, Class<?>> map) throws SQLException {
+        if (map == null || map.isEmpty()) {
+            return getArray(index, count);
+        }
+        throw new SQLFeatureNotSupportedException("getArray with non-empty Map not supported");
+    }
+
+    @Override
+    public ResultSet getResultSet() throws SQLException {
+        throw new SQLFeatureNotSupportedException("Array as ResultSet not supported");
+    }
+
+    @Override
+    public ResultSet getResultSet(Map<String, Class<?>> map) throws SQLException {
+        throw new SQLFeatureNotSupportedException("Array as ResultSet not supported");
+    }
+
+    @Override
+    public ResultSet getResultSet(long index, int count) throws SQLException {
+        throw new SQLFeatureNotSupportedException("Array as ResultSet not supported");
+    }
+
+    @Override
+    public ResultSet getResultSet(long index, int count, Map<String, Class<?>> map) throws SQLException {
+        throw new SQLFeatureNotSupportedException("Array as ResultSet not supported");
+    }
+
+    @Override
+    public void free() throws SQLException {
```

Once this method is called, all other methods should return an exception which is not the case.
There are no side-effects however this is the mandatory behavior described by the spec:

```
 * After <code>free</code> has been called, any attempt to invoke a
 * method other than <code>free</code> will result in a <code>SQLException</code>
 * being thrown.  If <code>free</code> is called multiple times, the subsequent
 * calls to <code>free</code> are treated as a no-op.
```

---

Right, on another look, that phrasing is mandatory. Check added.